src/Lecture3.hs

```diff
@@ -122,21 +132,24 @@ data Reward = Reward
     } deriving (Show, Eq)
 
 instance Semigroup Reward where
+  (<>) r1 r2 = Reward (Gold (gold r1 + gold r2)) (rewardSpecial r1 || rewardSpecial r2)
+    where gold = unGold . rewardGold
```

Comment:
Since `Gold` already implements a `Semigroup` instance, you can reuse it here
```suggestion
  (<>) r1 r2 = Reward (rewardGold r1 <> rewardGold r2) (rewardSpecial r1 || rewardSpecial r2)
```